triage/berghelroach/berghelroach.go

```diff
@@ -0,0 +1,312 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+Ported from Java com.google.gwt.dev.util.editdistance, which is:
+Copyright 2010 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License"); you may not
+use this file except in compliance with the License. You may obtain a copy of
+the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+License for the specific language governing permissions and limitations under
+the License.
+*/
+
+package berghelroach
+
+import (
+	"k8s.io/test-infra/triage/utils"
+)
+
+// Dist takes two strings and returns the edit distance between them.
+// It is the only name exported from this package.
+func Dist(a string, b string, limit int) int {
```

Comment:
nit: 0 being a magic number isn't documented, you could consider exporting a `Dist(a,b string)` that does the magic thing

Reply:
Can you rephrase this? I don't quite understand

Reply:
The fact that the function behaves differently if `limit==0` isn't documented in the comment above this.  Since go doesn't really do default args, the idiomatic way I've seen is to write boilerplate functions that delegate down to one implementation, e.g. `Dist(a, b)` calls `Dist(a, b, len(a)+len(b))`.  The hackish way I've seen is to make the "default" arg a varargs `Dist(a,b string, limit ...int)` but I don't think that makes sense here

Reply:
How do I export both if there's no function overloading in Go?

Reply:
two names?

if you would be calling `Dist(a, b, 0)` explicitly then it should probably be two methods, one wrapping the other

Reply:
spiffxp suggested just modifying the comment above to explicitly define 0 behavior.